zstacklib/zstacklib/__init__.py

```
@@ -1 +1,20 @@
+# Fix: replace subprocess with subprocess32 (Python 3 backport) which does
+# not manipulate GC around fork(), eliminating the race entirely.
+if os.name == 'posix' and sys.version_info[0] < 3:
+    try:
+        import subprocess32
+        sys.modules['subprocess'] = subprocess32
+    except ImportError:
+        raise ImportError(
+            "Failed to import 'subprocess32'. This package is required on Python 2 "
+            "to avoid a GC-disabling race condition in subprocess.Popen (BPO-27448). "
+            "Please install it via: pip install subprocess32"
+        ) 
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)